### ISSUE TITLE
cannot return `lent` expression from conditionals like `case`

### ISSUE THREAD

**Issue by arnetheduck** [nim-lang/Nim#23949]
### Description

```nim
type A = object
  field: int

proc works(a: A): lent int =
  a.field

proc x(a: A): lent int =
  case true
  of true:
    a.field
  of false:
    a.field

proc y(a: A): lent int =
  case true
  of true:
    a.field
  of false:
    raiseAssert "invalid"
```

Both `x` and `y` should have no problem returning the address of `a.field` to the consumer since they are equivalent to `works` - the address of the expression as a whole can be derived from the branches since the expressions therein have an address that readily can be used for this purpose.

### Nim Version

2.0.8

### Current Output

```text
Error: expression has no address
```

**Comment by juancarlospaco** [comment]
!nim c

```nim
type A = object
  field: int

proc works(a: A): lent int =
  a.field

proc x(a: A): lent int =
  case true
  of true:
    a.field
  of false:
    a.field

proc y(a: A): lent int =
  case true
  of true:
    a.field
  of false:
    raiseAssert "invalid"
```

**Comment by Araq** [comment]
Yeah well, we need more borrow checking before we need more lent/var support.

**Comment by arnetheduck** [comment]
the lack of case support stands out though - `lent` works with `if`, and case is just an `if` with different name.. why would the language support one but not the other?

**Comment by Araq** [comment]
I had no idea it does work with `if`.

**Comment by ringabout** [comment]
> lent works with if

What is the `if` example?

`proc isAssignable*(owner: PSym, n: PNode): TAssignableResult =` doesn't seem to specialize for `if/case`

**Comment by arnetheduck** [comment]
er, oops, you're right - I just retested for `if` and it doesn't work either :/ it would be welcome though, ie `if a: x.field1 else: x.field2` are canonical examples where this would be useful.

Borrow checking sounds like something on the consuming side - it seems like this issue could remain open for the producing side regardless.
